### ISSUE TITLE
Transformers call to `PreTrainedModel.to` is reported as an error

### ISSUE THREAD

**Issue by john-0u0** [microsoft/pyright#10043]
**Describe the bug**

When using the Transformers package to train or evaluate some model, after loading a model with `from_pretrained` , the `PreTrainedModel` calling `.to(device)` is reported as an error. (in fact it's been years since I first observed this type error)

**Code or Screenshots**

```python
import torch
from transformers import BertModel

model = BertModel.from_pretrained("bert-base-uncased")

# (variable) model: PreTrainedModel

if torch.cuda.is_available():
    device = torch.device("cuda")
    model.to(device)

# Argument of type "device" cannot be assigned to parameter "self" of type "PreTrainedModel" in function "__call__"
#     "device" is not assignable to "PreTrainedModel"
# Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)
```

Package versions:
```bash
$ pip show torch transformers | grep Version
Version: 2.1.2
Version: 4.46.1
```

**VS Code extension or command-line**

As a VS Code extension, using the latest Pylance `2025.3.1` .

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

Pyright bases its static type analysis on the type information provided in libraries. In this case, the `transformer` library is missing type annotations, so pyright does its best to infer type information. 

Your options include:
1. Set `useLibraryCodeForTypes` to false in the pyright configuration. This tells pyright to treat the types of all symbols imported from untyped libraries as `Any`. This will eliminate the type error, but it will also eliminate all completion suggestions for the library.
2. Work with the maintainers of the `transformer` library to add proper type annotations.
3. Create (or find) a stub package that describes the types in the `transformer` library. For more information about stubs, refer to [this documentation](https://microsoft.github.io/pyright/#/type-stubs).
4. Disable type checking by setting `typeCheckingMode` to "off".
5. Suppress the individual error using a `# type: ignore` or `# pyright: ignore` comment on the line.

If you suspect that pyright is incorrectly applying static typing rules, please create an issue with a minimal stand-alone code sample — one that doesn't rely on third-party code. In this case, I've dug into the third-party code, and based on my cursory analysis, it appears to me that pyright is doing the right thing here.
